test/base.py

```diff
@@ -33,21 +35,67 @@
 from .extended_equality import is_equivalent
 
 # Fail tests that take longer than this
-TEST_TIMEOUT = os.environ.get("TEST_TIMEOUT", 60)
+TEST_TIMEOUT = int(os.environ.get("TEST_TIMEOUT", 60))
+# Use testtools by default as a (mostly) drop in replacement for
+# unittest's TestCase. This will enable the fixtures used for capturing stdout
+# stderr, and pylogging to attach the output to stestr's result stream.
+USE_TESTTOOLS = os.environ.get("QE_USE_TESTOOLS", "TRUE") == "TRUE"
 
 
-class QiskitExperimentsTestCase(QiskitTestCase):
+if USE_TESTTOOLS:
+
+    class BaseTestCase(testtools.TestCase):
+        """Base test class."""
+
+        # testtools maintains their own version of assert functions which mostly
+        # behave as value adds to the std unittest assertion methods. However,
+        # for assertEquals and assertRaises modern unittest has diverged from
+        # the forks in testtools and offer more (or different) options that are
+        # incompatible testtools versions. Just use the stdlib versions so that
+        # our tests work as expected.
+        assertRaises = unittest.TestCase.assertRaises
+        assertEqual = unittest.TestCase.assertEqual
+
+else:
+
+    class BaseTestCase(unittest.TestCase):
+        """Base test class."""
+
+        def useFixture(self, fixture):
+            """Shim so that useFixture can be called in subclasses
+
+            The actual fixture is not used when using unittest.
+            """
+
+
+# TODO: copy enforce_subclasses_call decorator?
```

Comment:
I am undecided about `enforce_subclasses_call`. `QiskitTestCase` used it to require that all subclasses  call `setUp`, `setUpClass`, `tearDown` and `tearDownClass`. I think it is nice, but it's in `qiskit.test`. Jake said it would be okay to move it out of `qiskit.test` and keep it in qiskit terra. We could also copy it to here, but it is a fair amount of code. We could also just try to catch someone not calling `setUp`, etc. in code review. I am interested to hear if someone else has a preference.